nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/tools/MultipletTool.java

```diff
@@ -805,65 +400,115 @@ void refresh() {
+        if (analyzer != null) {
+            if (activeMultiplet.isPresent()) {
+                Multiplets.findMultipletMidpoint(activeMultiplet.get()).ifPresent(ppmCenter -> {
+                    try {
+                        activeMultiplet.get().setGenericMultiplet();
+                        List<Multiplet> multiplets = analyzer.splitRegion(ppmCenter);
+                        if (!multiplets.isEmpty()) {
+                            activeMultiplet = Optional.of(multiplets.get(0));
+                        } else {
+                            activeMultiplet = Optional.empty();
+                        }
+                        updateMultipletField(false);
+                        chart.refresh();
+                    } catch (IOException ex) {
+                    }
+                });
+            }
+        }
+    }
+
     public void adjustRegion() {
-        getAnalyzer();
-        double ppm0 = chart.getVerticalCrosshairPositions()[0];
-        double ppm1 = chart.getVerticalCrosshairPositions()[1];
-        analyzer.removeRegion(ppm0, ppm1);
-        analyzer.addRegion(ppm0, ppm1);
-        try {
-            activeMultiplet = analyzer.analyzeRegion((ppm0 + ppm1) / 2);
-            updateMultipletField(false);
-            chart.refresh();
-        } catch (IOException ex) {
-            log.warn(ex.getMessage(), ex);
+        Analyzer analyzer = getAnalyzer();
+        Double ppm0;
+        Double ppm1;
+        if (chart.getCrossHairs().hasCrosshairState("||")) {
+            ppm0 = chart.getVerticalCrosshairPositions()[0];
+            ppm1 = chart.getVerticalCrosshairPositions()[1];
+        } else {
+            if (activeMultiplet.isPresent()) {
+                Multiplet multiplet = activeMultiplet.get();
+                var optRegion = analyzer.getRegion(multiplet.getCenter());
+                if (optRegion.isPresent()) {
+                    var region = optRegion.get();
+                    ppm0 = region.getRegionStart(0);
+                    ppm1 = region.getRegionEnd(0);
+                } else {
+                    double[] minMax = Multiplets.getBoundsOfMultiplet(multiplet,analyzer.getTrimRatio());
+                    ppm0 = minMax[0];
+                    ppm1 = minMax[1];
+                }
+            } else {
+                return;
+            }
+        }
+        if ((ppm0 != null) && (ppm1 != null)) {
+            analyzer.removeRegion(ppm0, ppm1);
+            analyzer.addRegion(ppm0, ppm1);
+            try {
+                activeMultiplet = analyzer.analyzeRegion((ppm0 + ppm1) / 2);
+                updateMultipletField(false);
+                chart.refresh();
+            } catch (IOException ex) {
+                log.warn(ex.getMessage(), ex);
+            }
         }
     }
 
+
     public void addRegion(DatasetRegion region) {
-        getAnalyzer();
+        Analyzer analyzer = getAnalyzer();
+        System.out.println("add region " + analyzer);
```

Comment:
Change to a log statement or remove
```suggestion
        log.info("add region " + analyzer);
```